app/javascript/components/admin/site_settings/Appearance.jsx

```diff
@@ -1,7 +1,36 @@
+/* eslint-disable react/jsx-props-no-spreading */
+
 import React from 'react';
+import { Form } from 'react-bootstrap';
+import Button from 'react-bootstrap/Button';
+import { useForm } from 'react-hook-form';
+import useUpdateSiteSetting
+  from '../../../hooks/mutations/admins/site_settings/useUpdateSiteSetting';
+import useSiteSettings from '../../../hooks/queries/admin/site_settings/useSiteSettings';
 
 export default function Appearance() {
+  const { register, handleSubmit } = useForm();
+  const updateSiteSetting = useUpdateSiteSetting('PrimaryColor');
+  const { data: siteSettings } = useSiteSettings();
+
   return (
-    <p>Appearance</p>
+    <>
+      <h6>Primary Color</h6>
+      <Form onSubmit={handleSubmit(updateSiteSetting.mutate)}>
```

Comment:
I'd go with making a `PrimaryColorForm` component that contains the form fields and expects the `value` to be provided from its parent.

The `Appearance` component will only be in charge of extracting the required settings from the API and provide it to each form and create their layout.

For the layout I'd go with bootstrap's Grid system `<Row></Row> <Col></Col>` unless its necessary to do otherwise.

And I'd use the Form framework when making the forms.

Reply:
I didn't put much time on frontend/UI - this will be for next PR. 

This is just a prototype to show how the theming would work and if you guys agree with the way it's done.

I'm sorry I should have specified, I will keep your comments in mind.

Reply:
Oh, makes sense.